TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/redCloseAuto.java

```diff
@@ -0,0 +1,113 @@
+package org.firstinspires.ftc.teamcode.opmodes;
+
+import com.qualcomm.robotcore.eventloop.opmode.Autonomous;
+import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;
+
+import org.firstinspires.ftc.teamcode.hardware.Drivebase;
+import org.firstinspires.ftc.teamcode.hardware.VisionCamera;
+import org.firstinspires.ftc.teamcode.utility.CubeSide;
+import org.firstinspires.ftc.teamcode.utility.FieldSide;
+
+//Names the autonomous on the control hub dropdown.
+@Autonomous(name = "pixelMarksAuto")
+public class redCloseAuto extends LinearOpMode {
+    @Override
+    //This method is needed for teleOp and auto to run at all.
+    public void runOpMode() {
+        //Runs after the robot has been initialized.
+        //Assign vars here and create instances of needed classes.
+
+        telemetry.setAutoClear(false);
+        telemetry.addData("Status", "initializing");
+        telemetry.update();
+
+
+        //Assign drivebase to a instance.
+        Drivebase driveBase = new Drivebase(hardwareMap, this::opModeIsActive);
+        //Assign the camera to a instance of the object.
+        VisionCamera visionCamera = new VisionCamera(hardwareMap, FieldSide.RedClose);
+
+        //Needed for both types of opModes.
+        waitForStart();
+        //Runs while the opmode is running.
+        //This is needed to make a auto or teleOp run.
+        CubeSide side;
+        while ((side = visionCamera.getCubePrediction()) == null) sleep(20);
```

Comment:
It's generally preferred to add the { } to a single line while statement, just to future proof if you ever have to add more in the future. I would recommend at the very least adding a comment that says why this loop is here, i.e. //Scan for a cube locaiton every 20 ms until you detect a cube region on the left, center, or right.